src/plugins/numberline/components/editable-numberline-value.tsx

```diff
@@ -0,0 +1,103 @@
+import { observer } from 'mobx-react';
+import React, { useState, useRef, useEffect } from 'react';
+
+import "./numberline-tile.scss";
+
+interface IEditableValueProps {
+  readOnly?: boolean;
+  isTileSelected: boolean;
+  value: number;
+  offset: number;
+  minOrMax: "min" | "max";
+  onValueChange: (newValue: number) => void;
+}
+
+export const EditableNumberlineValue: React.FC<IEditableValueProps> = observer(function NumberlineTile(props) {
+  const { readOnly, isTileSelected, value, offset, minOrMax, onValueChange } = props;
+
+  const [isEditing, setIsEditing] = useState(false);
+  const inputRef = useRef<HTMLInputElement | null>(null);
+
+  useEffect(() => {
+    if (isEditing && inputRef.current) {
+      inputRef.current.select();
+    }
+  }, [isEditing]);
+
+  const handleClick = () => {
+    if (!readOnly && !isEditing) {
+      setIsEditing(true);
+      // Check if the input element exists before focusing and selecting
+      if (inputRef.current) {
+        inputRef.current.select();
+      }
+    }
+  };
+
+  const handleKeyDown = (e: React.KeyboardEvent<HTMLDivElement>) => {
+    let inputField = undefined;
+    let numberEntered = undefined;
+    const { key } = e;
+    switch (key) {
+      case "Enter":
+        inputField = (e.target as HTMLInputElement).value;
+        if (checkIfNumber(inputField)){
+          numberEntered = Number(inputField);
+          onValueChange(numberEntered);
+        }
+        setIsEditing(false);
+        break;
+      case "Escape":
+      case "Tab":
```

Comment:
Correct - that is what we actually want based on the ticket. Only thing that should save the value to the model (given that its non numeric) - is pressing "Enter" and unfocusing off the editable tile.

Reply:
In general UI terms, pressing tab is just a way to move focus away and as @tealefristoe says canceling on tab would be very surprising for users because it doesn't work that way anywhere else.

Reply:
Okay, gotcha. I'll remove the case "Tab" but keep the "Escape"